lib/zendesk_api/collection.rb

```diff
@@ -185,9 +185,8 @@ def fetch!(reload = false)
       elsif association && association.options.parent && association.options.parent.new_record?
         return (@resources = [])
       end
-
       @response = get_response(@query || path)
-      handle_response(@response.body)
+      handle_response(@response.body, (@query || path))
```

Comment:
To be honest I feel like this should be a separate method depending of the type of pagination we are using. So if we are using offset pagination, use the existing `handle_response` method. If we are using cursor pagination, use the new `handle_cursor_response` method.

Sending a param and bifurcating the execution path depending on it is an anti-pattern, as we are giving many responsibilities to one method.

We instead can create a new handle_cursor_response method and check which one to call here depending of the  (@query || path) value.

Reply:
Remember Sandy Metz talk about the Gilded Kata.